### PR TITLE
Revamp the User Guide

### LINKED ISSUE #123: [PE Dry Run]No restriction statement for phone, email in contacts add command in the help page
The help page does not include these info, it causes inconvenience when doing contacts add command.

<hr>

**Reported by:** @kelly9373
**Severity:** Not Specified

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#350]</sub>

### LINKED ISSUE #127: [PE Dry Run]No restriction statement for phone, email in the contacts edit command in the help page
<hr>

**Reported by:** @kelly9373
**Severity:** Not Specified

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#438]</sub>

### LINKED ISSUE #129: [PE Dry Run]No description of how to use contacts find command in detail in help page
The format only shows contacts find KEYWORD [MORE_KEYWORDS]. But what does KEYWORDS mean? Is it keyword of name or address or tag? So I tried contacts find [ADDRESS-KEYWORD] and it shows 0 persons listed. This makes confusion.

<hr>

**Reported by:** @kelly9373
**Severity:** Not Specified

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#527]</sub>

### LINKED ISSUE #139: [PE Dry Run]Please add more detailed instruction in the help page
<hr>

**Reported by:** @kelly9373
**Severity:** `Medium`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#1040]</sub>

### LINKED ISSUE #144: [PE Dry Run]Inconsistency in find command
image

tasks find is able to search by tags and other parameters but contacts find can only search by name 

<hr>

**Reported by:** @jiajunteo
**Severity:** `Low`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#1273]</sub>

### PR DESCRIPTION
**Pull request by indocomsoft**
To-Do: Add `clear` command

Fixes #123
Fixes #127 
Fixes #129
Fixes #139 
Fixes #144

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -29,174 +29,452 @@ apps.
...
 * Parameters can be in any order e.g. if the command specifies
 `n/NAME p/PHONE_NUMBER`, `p/PHONE_NUMBER n/NAME` is also acceptable.
+====
 
 [TIP]
 When `INDEX` is a parameter, it refers to the index of the item shown on the task or person list
 
+[NOTE]
+You can add multiple people or multiple tasks with the same name because internally we track them with
+a unique id.
```
Comment (taneliang) [inline docs/UserGuide.adoc]: I don't think we should talk about our implementation as this is targeted at users. We should still somehow justify why we're allowing "duplicate" people and tasks. Something like "Don't worry about having friends or tasks with the same name -- just add them and Cow will process them correctly."?

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -29,174 +29,452 @@ apps.
...
-This will display all tasks assigned to the contact in the tasks pane.
+****
+* Displays all tasks assigned to the person at the specified `INDEX`.
+* The index refers to the index number shown in the displayed person list.
+* The index *must be a positive integer* 1, 2, 3, ...
+****
+
+[TIP]
+The person specified at `INDEX` will be highlighted, and the tasks assigned are
+displayed on the displayed tasks list
```
Comment (taneliang) [inline docs/UserGuide.adoc]: I think the tips need to end with periods. This issue probably exists throughout the UG - there are some with periods and some without.

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -29,174 +29,452 @@ apps.
...
+* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
+* You can specify `t/` to indicate that you want to clear all existing tags.
+* Date and time are regarded as one field, and after editing, start DateTime
+must be before end DateTime.
+****
+
+Examples:
+
+* `tasks edit 1 sd/20050108 st/1235` +
+Edits the start DateTime of the 1st task to be 8 January 2005, 12.35pm.
```
Comment (taneliang) [inline docs/UserGuide.adoc]: Should `DateTime` be `date and time` instead? Also, "12:35pm" with a colon is more standard

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -29,174 +29,452 @@ apps.
...
+* `task edit 2 n/Brush the cows t/` +
```
Comment (taneliang) [inline docs/UserGuide.adoc]: Should there be a tag name after the `t/`?

File: docs/UserGuide.adoc (hunk 1 above)
Comment (indocomsoft) [reply docs/UserGuide.adoc]: Great! I was looking for a better way to phrase this

File: docs/UserGuide.adoc (hunk 4 above)
Comment (indocomsoft) [reply docs/UserGuide.adoc]: Nope, this is for clearing existing tags.

Comment (taneliang) [reply docs/UserGuide.adoc]: Oops, didn't see this mentioned a few lines above.

**Pull request merged**